client/blocks/signup-form/index.jsx

```diff
@@ -305,6 +306,20 @@ class SignupForm extends Component {
 				}
 			}
 
+			// Catch this early for P2 signup flow.
+			if (
+				this.props.isP2Flow &&
+				fields.username &&
+				fields.password &&
+				fields.username === fields.password
+			) {
+				messages = Object.assign( {}, messages, {
+					password: {
+						invalid: this.props.translate( 'Your password cannot be the same as your username.' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **55** times:
`translate( 'Your password cannot be the same as your username. Please pick a different password.' )` *ES Score: 7*

Reply:
Done.